cilium/CFP-39876-clustermesh-filtered-export.md

```diff
@@ -0,0 +1,153 @@
+
+Full support for **cross-cluster communication** and **network policy enforcement** is available **only when both the source (client) and destination (server) pods**—which may optionally be accessed via a service—**are part of a _global namespace_**.
+
+### Behavior When Pods Are Not in Global Namespaces
+
+If either the source or the destination pod does **not** belong to a global namespace, cross-cluster communication **may still succeed**, but under certain conditions although we do not support as part of the new mode. The evaluation of network policy for different cases is summarize below:
+
+
+### Summary of Enforcement Logic
+
+| Source Pod in Global NS | Destination Pod in Global NS | Network Policy Applies | Traffic Allowed? |
+|-------------------------|-------------------------------|-------------------------|------------------|
+| Yes                     | Yes                           | Yes                     | ✅ Enforced normally |
+| No                      | Yes                           | Ingress Policy          | ❌ Not Supported  |
+| Yes                     | No                            | Egress Policy           | ❌ Not Supported |
+| No                      | No                            | Any Policy              | ❌ Not Supported  |
+| Any                     | Any                           | No Policies             | ✅ Allowed (no restrictions) |
+
+## Global Service Support
+In the scoped-export mode, all services within global namespaces would automatically be marked as global/local depending on the provided annotation on the namespace. Users are not required to explicitly annotate individual services. Any service which is not inside a global namespace is considered local even if it has an associated global annotation. Such a service will not share backends with remote clusters in any case.  
+
+## MCS support
+Similar to global services, MCS support is only available for services created under global namespaces. 
+
+## Implementation details
+
+The proposed scoped-export mode has a config that users can pass in to mark all non local annotated namespaces as global. 
+- scoped-export-default-enabled - true/false Honors only the local namespace annotation and marks all non annotated namespaces as global if set to true. Honors the global annotation and marks all non annotated  namespaces as local if set to true
+
+#### Allow/Deny listing namespace
+To allowlist a namespace and mark it as global, users can annotate the namespace with:
+
+```
+clustermesh.cilium.io/global: "true"
+```
+
+The ClusterMesh API server will watch namespace resources. Adding this annotation to any namespace will result in the following steps:
+
+- A full synchronization is triggered for all resources within the newly annotated namespaces 
+- Only entries associated with namespaces explicitly annotated as global will be exported.
+- Unannotated namespaces are marked global/local depending on the value of config `scoped-export-default-enable `
```

Comment:
If we are maintaining the "global namespace" (vs an annotation having "export" in some form in its) I think the option should be named something like `clustermesh-default-global-namespace` for consistency with its annotation